File: GXpert/GXpert.Web/Modules/ServerTypes/Schools/SchoolStudentForm.ts
import { LookupEditor, ServiceLookupEditor, StringEditor, TextAreaEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface SchoolStudentForm {
    StudentId: LookupEditor;
    SchoolId: ServiceLookupEditor;
    ClassId: LookupEditor;
    Division: StringEditor;
    RollNumber: StringEditor;
    Description: TextAreaEditor;
    AcademicYearId: LookupEditor;
}

export class SchoolStudentForm extends PrefixedContext {
    static readonly formKey = 'Schools.SchoolStudent';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SchoolStudentForm.init)  {
            SchoolStudentForm.init = true;

            var w0 = LookupEditor;
            var w1 = ServiceLookupEditor;
            var w2 = StringEditor;
            var w3 = TextAreaEditor;

            initFormType(SchoolStudentForm, [
                'StudentId', w0,
                'SchoolId', w1,
                'ClassId', w0,
                'Division', w2,
                'RollNumber', w2,
                'Description', w3,
                'AcademicYearId', w0
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Schools/SchoolStudent/SchoolStudentForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Schools.Forms;

[FormScript("Schools.SchoolStudent")]
[BasedOnRow(typeof(SchoolStudentRow), CheckNames = true)]
public class SchoolStudentForm
{
    [HalfWidth]
    public int StudentId { get; set; }
    [HalfWidth]
    public int SchoolId { get; set; }
    [HalfWidth]
    public int ClassId { get; set; }
    [HalfWidth]
    public string Division { get; set; }
    [HalfWidth]
    public string RollNumber { get; set; }
    [TextAreaEditor(Rows =3)]
    public string Description { get; set; }
    [HalfWidth]
    public int AcademicYearId { get; set; }
}